src/main/java/com/google/flourbot/execution/MacroExecutionModuleImplementation.java

```diff
@@ -177,8 +191,14 @@ private String getDate(String pattern) {
     return myDateObj.format(myFormatObj);
   }
 
-  private String selectRandomFromData(List<String> data) {
+  private String selectRandomFromList(List<String> data) {
```

Comment:
this method doesn't really belong here. This a good reason to create SheetAction class and add protected methods that will execute this FIRST/LAST/RANDOM N logic.